samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -992,6 +997,18 @@ class SamzaContainer(
       }
     }
 
+    if (timerExecutor != null) {
+      info("Shutting down timer executor")
+      try {
+        timerExecutor.shutdown()
+        if(timerExecutor.awaitTermination(shutdownMs, TimeUnit.MILLISECONDS)) {
+          timerExecutor.shutdownNow()
+        }
+      } catch {
+        case e: Exception => error(e.getMessage, e)
```

Comment:
Log meaningful context in the error message instead of repeating e.getMessage. E.g., "Ignoring exception shutting down timer executor."